client/pages/StatUnits/View/StatUnitViewPage.jsx

```diff
@@ -1,75 +1,121 @@
 import tabs from './tabs/tabEnum'
 
-function StatUnitViewPage({
-  id,
-  type,
-  unit,
-  history,
-  historyDetails,
-  actions: {
-    fetchStatUnit,
-    navigateBack,
-    fetchHistory,
-    fetchHistoryDetails,
-    getUnitLinks,
-    getOrgLinks,
-    fetchSector,
-    fetchLegalForm,
-    fetchUnitStatus,
-    clear,
-  },
-  localize,
-}) {
-  const [activeTab, setActiveTab] = useState(tabs.main.name)
-  const [tabList, setTabList] = useState(Object.values(tabs))
+import { deleteStatUnit } from '../Search/actions.js'
 
-  useEffect(() => {
+class StatUnitViewPage extends React.Component {
+  static propTypes = {
+    id: oneOfType([number, string]).isRequired,
+    type: oneOfType([number, string]).isRequired,
+    unit: shape({
+      regId: number,
+      type: number.isRequired,
+      name: string.isRequired,
+      isDeleted: bool.isRequired,
+      changeReason: number.isRequired,
+      parentId: number,
+      address: shape({
+        addressLine1: string,
+        addressLine2: string,
+      }),
+    }),
+    history: shape({}),
+    historyDetails: shape({}),
+    actions: shape({
+      fetchStatUnit: func.isRequired,
+      fetchHistory: func.isRequired,
+      fetchHistoryDetails: func.isRequired,
+      fetchSector: func.isRequired,
+      fetchLegalForm: func.isRequired,
+      fetchUnitStatus: func.isRequired,
+      getUnitLinks: func.isRequired,
+      getOrgLinks: func.isRequired,
+      navigateBack: func.isRequired,
+      clear: func.isRequired,
+    }).isRequired,
+    localize: func.isRequired,
+  }
+
+  static defaultProps = {
+    unit: undefined,
+    history: undefined,
+    historyDetails: undefined,
+  }
+
+  state = { activeTab: tabs.main.name, tabList: Object.values(tabs) }
+
+  componentDidMount() {
+    const {
+      id,
+      type,
+      actions: { fetchStatUnit },
+    } = this.props
     fetchStatUnit(type, id)
-  }, [id, type, fetchStatUnit])
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.props.localize.lang !== nextProps.localize.lang ||
+      !R.equals(this.state, nextState) ||
+      !R.equals(this.props, nextProps)
+    )
+  }
 
-  useEffect(() => {
-    clear()
-  }, [clear])
+  componentWillUnmount() {
+    this.props.actions.clear()
+  }
 
-  const handleTabClick = (_, { name }) => {
-    setActiveTab(name)
+  handleTabClick = (_, { name }) => {
+    this.setState({ activeTab: name })
   }
 
-  const renderTabMenuItem = ({ name, icon, label }) => (
+  renderTabMenuItem = ({ name, icon, label }) => (
     <Menu.Item
       key={name}
       name={name}
-      content={localize(label)}
+      content={this.props.localize(label)}
       icon={icon}
-      active={activeTab === name}
-      onClick={handleTabClick}
+      active={this.state.activeTab === name}
+      onClick={this.handleTabClick}
     />
   )
 
-  const renderView = () => {
+  handleDeleteStatUnit = (statUnitType, statUnitRegId) => {
```

Comment:
Why do you add the redirect by `handleDeleteStatUnit` here?
Why don't you just call the function `deleteStatUnit` directly?